main.py

```diff
@@ -55,16 +55,40 @@ def unpickle(file):
 class NeuralNetwork(nn.Module):
   def __init__(self):
     super(NeuralNetwork, self).__init__()
-    self.flatten = nn.Flatten()
+    self.conv1 = nn.Conv2d(3, 1, 3)
+    self.Flatten = nn.Flatten()
+    self.pool = nn.MaxPool2d(2, 1)
+    self.conv2 = nn.Conv2d(1, 1, 3)
     self.linearReLUStack = nn.Sequential(
-      nn.Linear(3*32*32,512),
+      nn.Flatten(),
+      nn.Linear(28*28,512),
+      nn.ReLU(),
+      nn.Linear(512,512),
+      nn.ReLU(),
+      nn.Linear(512,512),
       nn.ReLU(),
       nn.Linear(512,512),
       nn.ReLU(),
       nn.Linear(512,10),
     )
+      # nn.Unflatten(32*32*3,(32,32,3)),
+      # nn.Conv1d(3, 1, 2),
+      # nn.Flatten(),
+      # nn.Linear(32*32,512),
+      # nn.ReLU(),
+      # nn.Linear(512,512),
+      # nn.ReLU(),
+      # nn.Linear(512,10),
   def forward(self,x):
-    logits = self.linearReLUStack(x)
+    logits = torch.reshape(x, (32,32,3))
+    logits = logits.permute(2,0,1)
+    logits = self.conv1(logits)
```

Comment:
It looks like your using convolutions without pooling in between, as well as not adding enough convolutiopns for the image features to be extracted. Normally, A typical CNN looks like: Cnv2D -> MaxPool -> Conv2D -> Max Pool -> Flatten -> Dense layers with activation -> final softmax or sigmoid layer. I would try (1) adding max pooling  (I see it in the class initialization but not the forward() method), and (2) tweaking the CNN parameters as having only 1 dim in the argument  (e.g for your second convolution ) will not do much! Here is a pretty helpful article going over CNN architecture https://medium.com/thecyphy/train-cnn-model-with-pytorch-21dafb918f48 with sample code, you don't need to follow this - it just has some helpful tips about how a typical CNN should look like et cetera!